### ISSUE TITLE
Add ability to destroy/detach interaction without explicit reference to parent Component

### ISSUE THREAD

**Issue by drglitch** [palantir/plottable#2886]
Interactions already know what parent they are attached to.
It would be great to either allow to call `.detach()` without a ref to parent plot to simply detach an interaction or create a `.destroy()` method that also automatically detaches and frees the instance.

**Comment by jtlan** [comment]
You mean on the `Interaction` itself? It does seem odd that `detachFrom()` takes a `Component` ... and then does nothing with it (this was originally added in the event future `Interaction`s could connect to multiple `Component`s). We should probably add a `detach()` call that just disconnects the `Interaction`.

**Comment by drglitch** [comment]
@jtlan yes - my thoughts exactly. a generic `detach()` that just gets rid of it would be ideal.
